hpctestlib/apps/amber/__init__.py

```diff
@@ -0,0 +1,124 @@
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+import reframe.utility.typecheck as typ
+
+__all__ = ["Amber_NVE"]
+
+
+class Amber_NVE(rfm.RunOnlyRegressionTest, pin_prefix=True):
+    '''Base class for the Amber Test.
+
+    Amber is a suite of biomolecular simulation programs. It
+    began in the late 1970's, and is maintained by an active
+    development community (see ambermd.org).
+
+    The presented abstract run-only class checks the work of amber.
+    To do this, it is necessary to define in tests the name
+    of the running script (input file), the output file,
+    as well as set the reference values of energy and possible
+    deviations from this value. This data is used to check if
+    the task is being executed correctly, that is, the final energy
+    is correct (approximately the reference). The default assumption
+    is that Amber is already installed on the device under test.
+    '''
+
+    #: Amber output file.
+    #:
+    #: :default: : 'amber.out'
+    output_file = variable(str, value='amber.out')
+
+    #: Amber input file.
+    #:
+    #: :default: : 'amber.out'
+    input_file = variable(str)
+
+    #: Reference value of energy, that is used for the comparison
+    #: with the execution ouput on the sanity step. Final value of
+    #: energy should be approximately the same
+    #:
+    #: :default: :class:`required`
+    energy_value = variable(float)
+
+    #: Maximum deviation from the reference value of energy,
+    #: that is acceptable.
+    #:
+    #: :default: :class:`required`
+    energy_tolerance = variable(float)
+
+    executable_files = parameter([
```

Comment:
I would still name this `platform`. I think it's more descriptive.
```suggestion
    # Parameter pack containing the platform ID, input file and executable.
    platform = parameter([
```